syntax = "proto3";

package sync;

service SyncService {
  rpc Health(Empty) returns (Empty) {}
  rpc SetLeader(Leader) returns (Empty) {}

  rpc SyncAddUser(SyncAddUserRequest) returns (Empty) {}
  rpc SyncDeleteUser(SyncDeleteUserRequest) returns (Empty) {}

  rpc SyncAddUnreadMessage(SyncAddMessage) returns (Empty) {}
  rpc SyncRemoveUnreadMessage(SyncAddMessage) returns (Empty) {}
  rpc SyncAddReadMessage(SyncRemoveMessage) returns (Empty) {}
  rpc SyncRemoveReadMessage(SyncRemoveMessage) returns (Empty) {}
}

message Empty {}

message Message {
  int32 id = 1;
  string sender = 2;
  string content = 3;
}

message Leader {
  int32 leader = 1;
}

message SyncAddUserRequest {
  string username = 1;
  string password = 2;
  string salt = 3;
}

message SyncDeleteUserRequest {
  string username = 1;
}

message SyncAddMessage {
  string user = 1;
  Message message = 2;
}

message SyncRemoveMessage {
  string user = 1;
  int32 message_id = 2;
}


